src/client/datascience/commands/activeEditorContext.ts

```diff
@@ -146,6 +148,8 @@ export class ActiveEditorContextService implements IExtensionSingleActivationSer
         this.updateMergedContexts();
     }
     private updateNativeNotebookContext() {
+        this.hasNativeNotebookOpen.set(this.vscNotebook.notebookDocuments.length > 0).ignoreErrors();
```

Comment:
There is actually an issue here, I believe, and it's a bit of an issue with exiting code as well, so I want to ping @DonJayamanne to check me on this. It looks like in spots like this function both with the added code here and with the existing code that we are looking at the vscNotebook interface. But that interface is just a wrapper around the VS Code interface. We'd be updating our can run above or notebook open context if a native notebook for something like the Markdown notebooks, or .NET Interactive notebooks are open. I think that we might want to watch some events off of the VS Code interface, but for checking a context like this we'd want to be looking at the active .editors off of our Editor provider (notebookEditorProvider) and then checking for any that were of type === 'native'. That would restrict it to just the notebook files that we know about and deal with. And it looks like we should make that change at more places in this file.

Reply:
Done

Reply:
Oops, i missed this message

Reply:
Yes, i agree with @IanMatthewHuff